src/util.h

```diff
@@ -32,6 +32,21 @@ uint64_t mg_now(void);     // Return milliseconds since Epoch
 #define MG_IPADDR_PARTS(ADDR) \
   MG_U8P(ADDR)[0], MG_U8P(ADDR)[1], MG_U8P(ADDR)[2], MG_U8P(ADDR)[3]
 
+#define MG_REG(x) ((volatile uint32_t *) (x))[0]
+#define MG_BIT(x) (((uint32_t) 1U) << (x))
+#define MG_SET_BITS(R, CLRMASK, SETMASK) (R) = ((R) & ~(CLRMASK)) | (SETMASK)
+
+#define MG_ROUND_UP(x, a) ((a) == 0 ? (x) : ((((x) + (a) -1) / (a)) * (a)))
+#define MG_ROUND_DOWN(x, a) ((a) == 0 ? (x) : (((x) / (a)) * (a)))
+
+#ifdef __GNUC__
+#define MG_ARM_DISABLE_IRQ() asm volatile ("cpsid i" : : : "memory")
+#define MG_ARM_ENABLE_IRQ() asm volatile ("cpsie i" : : : "memory")
+#else
+#define MG_ARM_DISABLE_IRQ()
+#define MG_ARM_ENABLE_IRQ()
```

Comment:
Definitely. Though I am not really sure that flashing need interrupts disabled. Disabling them however gets rid of weird lock-ups during writes.

Reply:
That is MCU-dependent. Some of them have two independent banks and some even can somehow read instructions while being written. Most of them can sort of do that, but flash reads are held while writings are taking place, so everything slows down to a halt. In those cases, a RAM flasher needs to be written and the rest of the system halted (IRQs will want to go to where the vector points them, either the old one or the new one, and depending on the order in which you write flash, there may or may not be code there...)

Reply:
H5 should work, you switch banks at the end. H7 I didn't see its architecture. F7 won't work that easy.

Reply:
We should have "a driver", and that should take care of if and how to disable interrupts. Then, we should probably use CMSIS intrinsics, to avoid reinventing CMSIS, they already move things like __dint() or similar to proper assembly for the compiler in use, detected at build time.
Yeah, we should do that, providing our ARM code will be conditionally left out when building for other arch (like Windoze)